PocketKit/Tests/SharedPocketKitTests/URLExtensionsTests.swift

```diff
@@ -0,0 +1,42 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import XCTest
+import SharedPocketKit
+
+class URLExtensionsTests: XCTestCase {
+    func test_withoutPercentEncoding_isValidURL_returnsURL() {
+        let source = "https://example.com/foo bar"
+        let url = URL(percentEncoding: source)
+        XCTAssertNotNil(url)
+        XCTAssertEqual(url!.absoluteString, "https://example.com/foo%20bar")
+    }
+
+    func test_withoutPercentEncoding_isInvalidURL_addsPercentEncodingAndReturnsURL() {
+        let source = "https://example.com/foo bar"
+        let url = URL(percentEncoding: source)
+        XCTAssertNotNil(url)
+        XCTAssertEqual(url!.absoluteString, "https://example.com/foo%20bar")
+    }
+
+    func test_withPercentEncoding_isValidURL_returnsURL() {
+        let source = "https://example.com/foo%20bar"
+        let url = URL(percentEncoding: source)
+        XCTAssertNotNil(url)
+        XCTAssertEqual(url!.absoluteString, "https://example.com/foo%20bar")
+    }
+
+    func test_withPercentEncoding_isInvalidURL_returnsURL() {
```

Comment:
```suggestion
    func test_withPercentEncoding_isInvalidURL_returnsNil() {
```

Reply:
Good catch, I'll make the fix in the SavedItem-related PR